src/feature.js

```diff
@@ -65,59 +74,55 @@ class FxABrowserFeature {
   }
 
   async updateState() {
-
     // The stored sessionToken will always be the source of truth when checking
     // account state.
     const user = await browser.fxa.getSignedInUser();
-    if (!user) {
-      this._noUser();
-    }
-
-    if (user && !user.verified) {
-      this._unverifiedUser();
-    }
-
-    if (user && user.verified) {
-      this._verifiedUser(user);
+    if (user) {
+      if (user.verified) {
+        this.verifiedUser(user);
+      } else {
+        this.unverifiedUser();
+      }
+    } else {
+      this.noUser();
     }
   }
 
-  _noUser() {
-    this._defaultAvatar();
-    browser.browserAction.setIcon({ path: "icons/avatar.svg" });
+  noUser() {
```

Comment:
Fwiw I ditched the leading `_` on these "private" methods to resolve a naming clash between the property `this._userAvatar` and the method `this._userAvatar()`.